LUSDChickenBonds/src/NFTArtwork/ChickenOutArtwork.sol

```diff
@@ -0,0 +1,213 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.10;
+
+import "./BondNFTArtworkBase.sol";
+import "./ChickenOutGenerated.sol";
+import "./ChickenOutTraitWeights.sol";
+
+contract ChickenOutArtwork is BondNFTArtworkBase, ChickenOutGenerated, ChickenOutTraitWeights {
+    constructor(
+        BondNFTArtworkCommon _common,
+        ChickenOutGenerated1 _g1
+    )
+        BondNFTArtworkBase(_common)
+        ChickenOutGenerated(_g1)
+    {}
+
+    ///////////////////////////////////////
+    // Abstract function implementations //
+    ///////////////////////////////////////
+
+    function _tokenURIImplementation(CommonData memory _commonData)
+        internal
+        view
+        virtual
+        override
+        returns (string memory)
+    {
+        ChickenOutData memory chickenOutData;
+        _calcChickenOutData(_commonData, chickenOutData);
+
+        return _getMetadataJSON(
+            _commonData,
+            _getSVG(_commonData, chickenOutData),
+            _getMetadataExtraAttributes(_commonData, chickenOutData)
+        );
+    }
+
+    ///////////////////////
+    // Private functions //
+    ///////////////////////
+
+    function _getObjectRainbowGradientUrl(string memory _tokenIDString)
+        private
+        pure
+        returns (bytes memory)
+    {
+        return abi.encodePacked(
+            'url(#co-chicken-', _tokenIDString, '-object-rainbow-gradient)'
+        );
+    }
+
+    function _getObjectFill(EggTraitWeights.ShellColor _color, string memory _tokenIDString)
+        private
+        pure
+        returns (bytes memory)
+    {
+        return (
+            _color == EggTraitWeights.ShellColor.Rainbow     ?
+                _getObjectRainbowGradientUrl(_tokenIDString) :
+            _color == EggTraitWeights.ShellColor.Luminous    ?
+                bytes('#e5eff9')                             :
+            // default
+                bytes(_getSolidObjectColor(_color))
+        );
+    }
+
+    function _getObjectStyle(EggTraitWeights.ShellColor _color, string memory _tokenIDString)
+        private
+        pure
+        returns (bytes memory)
+    {
+        return abi.encodePacked(
+            'fill:',
+            _getObjectFill(_color, _tokenIDString),
+            _color == EggTraitWeights.ShellColor.Luminous ? ';mix-blend-mode:luminosity' : ''
+        );
+    }
+
+    function _calcChickenOutData(
+        CommonData memory _commonData,
+        ChickenOutData memory _chickenOutData
+    )
+        private
+        view
+    {
+        uint80 dna = _commonData.finalHalfDna;
+
+        _chickenOutData.chickenColor = _getChickenColor(_cutDNA(dna, 0, 80), _commonData.shellColor);
```

Comment:
No big deal, but do we need `_cutDNA` here?

Reply:
Yes, it's being used to normalize the DNA to the [0,1) range, as expected by `_getChickenColor()`. There are simpler ways to accomplish that, but `_cutDNA()` has been battletested 😄